### PR TITLE
feat(RCTTextBox): TextInput functionality for UWP

### LINKED ISSUE #96: Fit-and-finish work for ReactTextInputManager

### PR DESCRIPTION
**Pull request by rozele**
This cleans up a lot of the behavior for TextInput, but not all of it. The properties that seem to be working are:
* autoCorrect
* keyboardType
* maxLength
* onFocus/onBlur
* onSelectionChange
* onEndEditing
* onSubmitEditing
* clearTextOnFocus
* selectTextOnFocus
* placeholder
* selectionColor
* editable

The properties that are not working are:
* numberOfLines
* password
* placeholderTextColor
* autoCapitalize

We do not yet support formatted children in the TextInput, which will likely require the use of RichEditBox and RTF. The multiline kind of works, but really only when you do some auto-resizing. The focus and blur methods are not working either.

Fixes #96

### CODE REVIEW

File: ReactWindows/ReactNative/Views/Text/ReactTextShadowNode.cs [hunk 1]
```diff
@@ -289,70 +308,48 @@ private static MeasureOutput MeasureText(CSSNode node, float width, float height
...
-            if (textNode._fontFamily != null)
-            {
-                var fontFamily = new FontFamily(textNode._fontFamily);
-                inline.FontFamily = fontFamily;
-            }
-
-            return inline;
         }
     }
 }
```
Comment (ebragge) [inline ReactWindows/ReactNative/Views/Text/ReactTextShadowNode.cs]: Missing empty line

File: ReactWindows/ReactNative/Views/TextInput/ReactTextBox.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            }
+
+            if (SelectTextOnFocus)
+            {
+                SelectionStart = 0;
+                SelectionLength = Text.Length;
+            }
+        }
+
+        private void OnLayoutUpdated(object sender, object e)
```
Comment (ebragge) [inline ReactWindows/ReactNative/Views/TextInput/ReactTextBox.cs]: OnLayoutUpdated(object sender, **RoutedEventArgs** e) ?

File: ReactWindows/ReactNative/Views/TextInput/ReactTextInputSubmitEditingEvent.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            var eventData = new JObject
+            {
+                { "target", ViewTag },
+                { "text", _text },
+            };
+
+            eventEmitter.receiveEvent(ViewTag, EventName, eventData);
+        }
+    }
+}
```
Comment (ebragge) [inline ReactWindows/ReactNative/Views/TextInput/ReactTextInputSubmitEditingEvent.cs]: Empty line missing

File: ReactWindows/ReactNative/Views/TextInput/ReactTextInputSelectionEvent.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+            var eventData = new JObject
+            {
+                { "selection", selectionData },
+            };
+
+            eventEmitter.receiveEvent(ViewTag, EventName, eventData);
+        }
+    }
+}
```
Comment (ebragge) [inline ReactWindows/ReactNative/Views/TextInput/ReactTextInputSelectionEvent.cs]: Empty line missing

**Comment by ebragge** [comment]
LGTM

File: ReactWindows/ReactNative/Views/TextInput/ReactTextBox.cs (hunk 2 above)
Comment (rozele) [inline ReactWindows/ReactNative/Views/TextInput/ReactTextBox.cs]: The LayoutUpdated event delegate type is EventHandler<System.Object>.

---
In reply to: [56671657](https://github.com/CatalystCode/react-native/pull/273#discussion_r56671657) [](ancestors = 56671657)

File: ReactWindows/ReactNative/Views/TextInput/ReactTextInputSubmitEditingEvent.cs (hunk 3 above)
Comment (rozele) [inline ReactWindows/ReactNative/Views/TextInput/ReactTextInputSubmitEditingEvent.cs]: Thanks.

---
In reply to: [56673153](https://github.com/CatalystCode/react-native/pull/273#discussion_r56673153) [](ancestors = 56673153)

File: ReactWindows/ReactNative/Views/Text/ReactTextShadowNode.cs (hunk 1 above)
Comment (rozele) [inline ReactWindows/ReactNative/Views/Text/ReactTextShadowNode.cs]: Got it!

---
In reply to: [56670447](https://github.com/CatalystCode/react-native/pull/273#discussion_r56670447) [](ancestors = 56670447)

**Pull request merged**